### PR TITLE
feat(refactor-helpers): refactor helper classes made them static inst…

### PR DESCRIPTION
**Pull request by andCastStatic**
…ead of components and added unit testing for them

### CODE REVIEW

File: service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreator.java [hunk 1]
```diff
@@ -17,16 +17,25 @@
...
+ */
 public class PolyResponseEntityCreator<O extends BaseServiceResponse> {
 
-    public ResponseEntity<List<O>> create(Page<O> page, HttpServletResponse httpServletResponse) {
+    /**
+     * Creates a Poly ResponseEntity with pagination
+     * @param page
+     * @param httpServletResponse
+     * @return ResponseEntity
+     * @param <O>
```
Comment (guttff) [inline service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreator.java]: return type should be last

File: service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreator.java [hunk 2]
```diff
@@ -41,7 +50,13 @@ public ResponseEntity<List<O>> create(Page<O> page, HttpServletResponse httpServ
         return responseEntity;
     }
 
-    private void setHeadersInResponse(final Page<O> page, final HttpServletResponse httpServletResponse) {
+    /**
+     * Creates pagination header
+     * @param page
+     * @param httpServletResponse
+     * @param <O>
+     */
```
Comment (guttff) [inline service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreator.java]: the order matters

Comment (guttff) [reply service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreator.java]: the java doc seem to be wrong

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/CreateResponseEntityCreatorTest.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class CreateResponseEntityCreatorTest {
```
Comment (whoswendy) [inline service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/CreateResponseEntityCreatorTest.java:32]: add javadoc please

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/CreateResponseEntityCreatorTest.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    @Test
+    public void test_create_responseEntity() {
+        ResponseEntity<BaseServiceResponseTest> response = CreateResponseEntityCreator.create(new BaseServiceResponseTest("test", "test"));
+        assertNotNull(response);
+        assertEquals(201, response.getStatusCodeValue());
+        assertEquals(HttpStatus.CREATED, response.getStatusCode());
+    }
+
+}
```
Comment (whoswendy) [inline service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/CreateResponseEntityCreatorTest.java]: add new line eof please

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/MonoResponseEntityCreatorTest.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    @Test
+    public void test_create_responseEntity() {
+        var response = MonoResponseEntityCreator.create(new BaseServiceResponseTest("test", "test"));
+        assertNotNull(response);
+        assertEquals(200, response.getStatusCodeValue());
+        assertEquals("test", Objects.requireNonNull(response.getBody()).getValue());
+        assertEquals("test", response.getBody().getId());
+    }
+
+}
```
Comment (whoswendy) [inline service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/MonoResponseEntityCreatorTest.java]: add new line eof

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/MonoResponseEntityCreatorTest.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+import io.americanexpress.synapse.service.rest.controller.helpers.model.BaseServiceResponseTest;
+import org.junit.Test;
+
+import java.util.Objects;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class MonoResponseEntityCreatorTest {
```
Comment (whoswendy) [inline service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/MonoResponseEntityCreatorTest.java:28]: javadoc for test class

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreatorTest.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.data.domain.PageImpl;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+
+public class PolyResponseEntityCreatorTest {
```
Comment (whoswendy) [inline service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreatorTest.java:30]: add javadoc for test class

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreatorTest.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    public void test_create_responseEntity() {
+        var page = new PageImpl<>(List.of(new BaseServiceResponseTest("test", "test")));
+        var httpServletRequest = new MockHttpServletResponse();
+        var responseEntity = PolyResponseEntityCreator.create(page, httpServletRequest);
+        assertNotNull(responseEntity);
+        assertEquals(200, responseEntity.getStatusCodeValue());
+        assertEquals("test", responseEntity.getBody().get(0).getValue());
+        assertEquals("test", responseEntity.getBody().get(0).getId());
+    }
+}
```
Comment (whoswendy) [inline service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreatorTest.java]: add new line eof

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/model/BaseServiceResponseTest.java [hunk 9]
```diff
@@ -0,0 +1,48 @@
+package io.americanexpress.synapse.service.rest.controller.helpers.model;
+
+import io.americanexpress.synapse.service.rest.model.BaseServiceResponse;
+
+public class BaseServiceResponseTest extends BaseServiceResponse {
```
Comment (whoswendy) [inline service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/model/BaseServiceResponseTest.java:21]: add javadoc please

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveCreateResponseEntityCreatorTest.java [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpStatus;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class ReactiveCreateResponseEntityCreatorTest {
```
Comment (whoswendy) [inline service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveCreateResponseEntityCreatorTest.java:33]: javadoc please

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveCreateResponseEntityCreatorTest.java [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+
+
+    @Test
+    public void test_create_responseEntity() {
+        var response = ReactiveCreateResponseEntityCreator.create(new BaseServiceResponseTest("test", "test"));
+        assertNotNull(response);
+        assertEquals(201, response.getStatusCodeValue());
+        assertEquals(HttpStatus.CREATED, response.getStatusCode());
+    }
+}
```
Comment (whoswendy) [inline service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveCreateResponseEntityCreatorTest.java]: new line eof

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveMonoResponseEntityCreatorTest.java [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+import io.americanexpress.synapse.service.rest.controller.helpers.model.BaseServiceResponseTest;
+import org.junit.Test;
+import reactor.core.publisher.Mono;
+
+import java.util.Objects;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class ReactiveMonoResponseEntityCreatorTest {
```
Comment (whoswendy) [inline service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveMonoResponseEntityCreatorTest.java:30]: javadoc please

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveMonoResponseEntityCreatorTest.java [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+        var response = ReactiveMonoResponseEntityCreator.create(Mono.just(new BaseServiceResponseTest("test", "test")));
+        assertNotNull(response);
+        assertEquals(200, response.getStatusCodeValue());
+        Objects.requireNonNull(response.getBody())
+                .subscribe(results -> {
+                    assertEquals("test", results.getValue());
+                    assertEquals("test", results.getId());
+                });
+    }
+}
```
Comment (whoswendy) [inline service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveMonoResponseEntityCreatorTest.java]: new line eof

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactivePolyResponseEntityCreatorTest.java [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.data.domain.PageImpl;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+import java.util.List;
+import java.util.Objects;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+class ReactivePolyResponseEntityCreatorTest {
```
Comment (whoswendy) [inline service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactivePolyResponseEntityCreatorTest.java:30]: javadoc please

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactivePolyResponseEntityCreatorTest.java [hunk 15]
```diff
@@ -0,0 +1,29 @@
...
+        var responseEntity = ReactivePolyResponseEntityCreator.create(page, httpServletRequest);
+        assertNotNull(responseEntity);
+        assertEquals(200, responseEntity.getStatusCodeValue());
+        Objects.requireNonNull(responseEntity.getBody())
+                .subscribe(results -> {
+                    assertEquals("test", results.getValue());
+                    assertEquals("test", results.getId());
+                });
+    }
+}
```
Comment (whoswendy) [inline service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactivePolyResponseEntityCreatorTest.java]: new line eof

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/CreateResponseEntityCreatorTest.java [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+
+    @BeforeAll
+    public static void init() {
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        RequestContextHolder.setRequestAttributes(new ServletRequestAttributes(request));
+    }
+
+
+    @Test
+    public void test_create_responseEntity() {
```
Comment (whoswendy) [inline service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/CreateResponseEntityCreatorTest.java]: can you use the naming format methodName_givenSomething_expectedSomething ?
ex: create_givenServiceResponse_expectedResponseEntity

Comment (andCastStatic) [reply service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/CreateResponseEntityCreatorTest.java]: renamed all test methods to recommended.

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactivePolyResponseEntityCreatorTest.java (hunk 15 above)
Comment (andCastStatic) [reply service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactivePolyResponseEntityCreatorTest.java]: added

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactivePolyResponseEntityCreatorTest.java (hunk 14 above)
Comment (andCastStatic) [reply service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactivePolyResponseEntityCreatorTest.java:30]: added

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveMonoResponseEntityCreatorTest.java (hunk 13 above)
Comment (andCastStatic) [reply service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveMonoResponseEntityCreatorTest.java]: added

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveMonoResponseEntityCreatorTest.java (hunk 12 above)
Comment (andCastStatic) [reply service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveMonoResponseEntityCreatorTest.java:30]: added

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreatorTest.java (hunk 7 above)
Comment (andCastStatic) [reply service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreatorTest.java:30]: added

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/MonoResponseEntityCreatorTest.java (hunk 6 above)
Comment (andCastStatic) [reply service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/MonoResponseEntityCreatorTest.java:28]: added

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/MonoResponseEntityCreatorTest.java (hunk 5 above)
Comment (andCastStatic) [reply service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/MonoResponseEntityCreatorTest.java]: added

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/CreateResponseEntityCreatorTest.java (hunk 4 above)
Comment (andCastStatic) [reply service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/CreateResponseEntityCreatorTest.java]: added

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/CreateResponseEntityCreatorTest.java (hunk 3 above)
Comment (andCastStatic) [reply service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/CreateResponseEntityCreatorTest.java:32]: added

File: service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreator.java (hunk 2 above)
Comment (andCastStatic) [reply service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreator.java]: added

Comment (andCastStatic) [reply service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreator.java]: and fixed

File: service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreator.java (hunk 1 above)
Comment (andCastStatic) [reply service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreator.java]: fixed

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/CreateResponseEntityCreatorTest.java [hunk 17]
```diff
@@ -0,0 +1,41 @@
+package io.americanexpress.synapse.service.rest.controller.helpers;
```
Comment (whoswendy) [inline service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/CreateResponseEntityCreatorTest.java:14]: add copyright

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/MonoResponseEntityCreatorTest.java [hunk 18]
```diff
@@ -0,0 +1,29 @@
+package io.americanexpress.synapse.service.rest.controller.helpers;
```
Comment (whoswendy) [inline service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/MonoResponseEntityCreatorTest.java:14]: add copyright

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreatorTest.java [hunk 19]
```diff
@@ -0,0 +1,31 @@
+package io.americanexpress.synapse.service.rest.controller.helpers;
```
Comment (whoswendy) [inline service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreatorTest.java:14]: add copyright

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/model/BaseServiceResponseTest.java [hunk 20]
```diff
@@ -0,0 +1,48 @@
+package io.americanexpress.synapse.service.rest.controller.helpers.model;
```
Comment (whoswendy) [inline service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/model/BaseServiceResponseTest.java:14]: add copyright

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveCreateResponseEntityCreatorTest.java [hunk 21]
```diff
@@ -0,0 +1,38 @@
+package io.americanexpress.synapse.service.rest.controller.reactive.helpers;
```
Comment (whoswendy) [inline service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveCreateResponseEntityCreatorTest.java:14]: add copyright

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveMonoResponseEntityCreatorTest.java [hunk 22]
```diff
@@ -0,0 +1,31 @@
+package io.americanexpress.synapse.service.rest.controller.reactive.helpers;
```
Comment (whoswendy) [inline service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveMonoResponseEntityCreatorTest.java:14]: add copyright

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactivePolyResponseEntityCreatorTest.java [hunk 23]
```diff
@@ -0,0 +1,35 @@
+package io.americanexpress.synapse.service.rest.controller.reactive.helpers;
```
Comment (whoswendy) [inline service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactivePolyResponseEntityCreatorTest.java:14]: add copyright

**Review by whoswendy — COMMENTED** [review]
need to remove the autowired response creators from base controller classes since @Component is now removed

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/CreateResponseEntityCreatorTest.java (hunk 17 above)
Comment (andCastStatic) [reply service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/CreateResponseEntityCreatorTest.java:14]: added

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/MonoResponseEntityCreatorTest.java (hunk 18 above)
Comment (andCastStatic) [reply service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/MonoResponseEntityCreatorTest.java:14]: added

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/model/BaseServiceResponseTest.java (hunk 20 above)
Comment (andCastStatic) [reply service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/model/BaseServiceResponseTest.java:14]: added

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveMonoResponseEntityCreatorTest.java (hunk 22 above)
Comment (andCastStatic) [reply service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveMonoResponseEntityCreatorTest.java:14]: added

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactivePolyResponseEntityCreatorTest.java (hunk 23 above)
Comment (andCastStatic) [reply service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactivePolyResponseEntityCreatorTest.java:14]: added

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveCreateResponseEntityCreatorTest.java (hunk 21 above)
Comment (andCastStatic) [reply service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveCreateResponseEntityCreatorTest.java:14]: added

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreatorTest.java (hunk 19 above)
Comment (andCastStatic) [reply service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreatorTest.java:14]: added

**Review by whoswendy — APPROVED** [review]

File: service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/model/BaseServiceResponseTest.java [hunk 24]
```diff
@@ -0,0 +1,64 @@
...
+    /**
+     * value
+     */
+    private String value;
+
+    /**
+     * Constructor
+     * @param value
+     * @param id
+     */
```
Comment (guttff) [inline service/synapse-service-rest/src/test/java/io/americanexpress/synapse/service/rest/controller/helpers/model/BaseServiceResponseTest.java]: complete javadocs

File: service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/helpers/CreateResponseEntityCreator.java [hunk 25]
```diff
@@ -16,12 +16,14 @@
...
 import org.springframework.http.ResponseEntity;
-import org.springframework.stereotype.Component;
 import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
 
 import java.net.URI;
 
-@Component
+/**
+ * {@code CreateResponseEntityCreator} Helps creates ResponseEntity
+ * @param <O> BaseServiceResponse
```
Comment (guttff) [inline service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/helpers/CreateResponseEntityCreator.java]: complete javadoc

File: service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/helpers/MonoResponseEntityCreator.java [hunk 26]
```diff
@@ -16,12 +16,20 @@
 import io.americanexpress.synapse.service.rest.model.BaseServiceResponse;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.stereotype.Component;
 
-@Component
+/**
+ * {@code MonoResponseEntityCreator} Creates ResponseEntity for mono response
+ * @param <O> BaseServiceResponse
```
Comment (guttff) [inline service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/helpers/MonoResponseEntityCreator.java]: complete javadoc

File: service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreator.java [hunk 27]
```diff
@@ -17,16 +17,24 @@
...
+ * @param <O> a response extending {@link BaseServiceResponse}
+ */
 public class PolyResponseEntityCreator<O extends BaseServiceResponse> {
 
-    public ResponseEntity<List<O>> create(Page<O> page, HttpServletResponse httpServletResponse) {
+    /**
+     * Creates a Poly ResponseEntity with pagination
+     * @param page sets pagination
+     * @param httpServletResponse response
+     * @return ResponseEntity
```
Comment (guttff) [inline service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreator.java]:
```suggestion
     * @return {@link ResponseEntity}
```

File: service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreator.java [hunk 28]
```diff
@@ -17,16 +17,24 @@
...
-@Component
+/**
+ * {@code PolyResponseEntityCreator} creates ResponseEntity for poly responses.
+ * @param <O> a response extending {@link BaseServiceResponse}
+ */
 public class PolyResponseEntityCreator<O extends BaseServiceResponse> {
 
-    public ResponseEntity<List<O>> create(Page<O> page, HttpServletResponse httpServletResponse) {
+    /**
+     * Creates a Poly ResponseEntity with pagination
```
Comment (guttff) [inline service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreator.java]: add punctuation

File: service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreator.java [hunk 29]
```diff
@@ -41,7 +49,12 @@ public ResponseEntity<List<O>> create(Page<O> page, HttpServletResponse httpServ
         return responseEntity;
     }
 
-    private void setHeadersInResponse(final Page<O> page, final HttpServletResponse httpServletResponse) {
+    /**
+     * Creates pagination header
```
Comment (guttff) [inline service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/helpers/PolyResponseEntityCreator.java:53]: add puncutation

File: service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveCreateResponseEntityCreator.java [hunk 30]
```diff
@@ -14,16 +14,17 @@
...
-import org.springframework.stereotype.Component;
 import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+import reactor.core.publisher.Mono;
 
 import java.net.URI;
 
-@Component
+/**
+ * {@code ReactiveCreateResponseEntityCreator} Creates a ResponseEntity
+ * @param <O>
```
Comment (guttff) [inline service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveCreateResponseEntityCreator.java]: complete javadoc

File: service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveCreateResponseEntityCreator.java [hunk 31]
```diff
@@ -14,16 +14,17 @@
...
 import org.springframework.http.ResponseEntity;
-import org.springframework.stereotype.Component;
 import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+import reactor.core.publisher.Mono;
 
 import java.net.URI;
 
-@Component
+/**
+ * {@code ReactiveCreateResponseEntityCreator} Creates a ResponseEntity
```
Comment (guttff) [inline service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/reactive/helpers/ReactiveCreateResponseEntityCreator.java]:
```suggestion
 * {@code ReactiveCreateResponseEntityCreator} Creates a {@link ResponseEntity}.
```

**Review by guttff — APPROVED** [review]

**Pull request merged**